src/lib/fastDownward/fastDownward.ts

```diff
@@ -41,7 +48,7 @@ export function computeCost<LU extends LearningUnit>(
 		contextSwitchPenalty !== 1
 			? // Check if the current LU requires any skills that are provided by the LU of the currentNode, only if a penalty is defined
 			  lu.teachingGoals.some(
-					skill => currentNode.action?.requiredSkills.includes(skill) ?? true
+					skill => currentNode.action?.requiredSkills.extractSkills().includes(skill) ?? true
```

Comment:
Very good point, that I forgott in our talks.

But I ask myself how often variables are extracted from expressions at this point and if it could be avoided (by chachig the results per expression)...